src/openApi/apiRequestModel.ts

```diff
@@ -12,17 +12,21 @@ export interface ApiRequest<T> {
   };
   headers: Record<string, string>;
   pathParameters: Record<string, string>;
-  queryStringParameters?: any;
+  queryStringParameters?: Query;
 }
 
-export interface PostRequest<T> extends ApiRequest<T> {
-  body: T;
+export interface PostRequest<Body = any, Query = any> extends ApiRequest<Body, Query> {
+  body: Body;
```

Comment:
In theory, the body should still be optional as we have endpoints with an empty body. But then in the majority of controllers, we would have to null-check the body in each route. Instead, we can have it still required and pass `{}` if needed

Reply:
Who will need to pass `{}`? The HTTP clients of APIs?

Or is it "just" when writing unit tests or so?

Reply:
Ah no just when you define the controller method. You can even so `PostRequest<null | undefined>` or even `PostRequest` if the body is not used. The `PostRequest<{}>` wasn't the best example